core/src/ParaGridIO.cpp

```diff
@@ -272,7 +272,7 @@ void ParaGridIO::dumpModelState(
     for (auto entry : ModelArray::definedDimensions) {
         ModelArray::Dimension dim = entry.first;
         size_t dimSz = (dimCompMap.count(dim)) ? ModelArray::nComponents(dimCompMap.at(dim))
-                                               : dimSz = entry.second.globalLength;
+                                               : dimSz = entry.second.localLength;
```

Comment:
found it :sweat: 

`localLength` should be `globalLength`. This part sets the total dimension of the array ready to be written to file. Only the portion of ModelArray will actually be written, but netcdf parallel needs to know the total size so it can write to the correct portion.

```suggestion
                                               : dimSz = entry.second.globalLength;
```

Reply:
I tested locally and this fixes the issue with the test

Reply:
I made this change because a 1 day example run was failing with the following error message. Making the change to `localLength` let the file writing finish correctly.
```
2010-01-01T23:45:00Z
libc++abi: terminating due to uncaught exception of type netCDF::exceptions::NcEdge: NetCDF: Start+count exceeds dimension bound
file: /tmp/netcdf-cxx-20250303-65682-ahp603/netcdf-cxx4-4.3.1/cxx4/ncVar.cpp  line:1049
Abort trap: 6
```